src/luxonis_ml/embeddings/embedding_utils.py

```diff
@@ -0,0 +1,100 @@
+"""
+Embeddings Extractor and Storage
+
+This module provides utility functions for extracting embeddings from both PyTorch and ONNX models, 
+and subsequently storing and retrieving these embeddings from disk.
+
+Functions:
+    - extract_embeddings(model, data_loader): 
+      Extracts embeddings from a given PyTorch model using data from a specified DataLoader.
+      
+    - extract_embeddings_onnx(ort_session, data_loader, output_layer_name):
+      Extracts embeddings from a specified ONNX model (provided as an ONNX Runtime session) 
+      using data from a specified DataLoader. Allows targeting a specific output layer for extraction.
+      
+    - save_embeddings(embeddings, labels, save_path):
+      Saves both embeddings and their associated labels to the disk at a given path.
+      
+    - load_embeddings(save_path):
+      Loads embeddings and their associated labels from the disk at a given path.
+
+Usage Examples:
+    1. Extract embeddings from a PyTorch model:
+        embeddings, labels = extract_embeddings(pytorch_model, data_loader)
+    
+    2. Extract embeddings from an ONNX model:
+        ort_session = ort.InferenceSession('model.onnx')
+        embeddings, labels = extract_embeddings_onnx(ort_session, data_loader, "/Flatten_output_0")
+    
+    3. Save embeddings to disk:
+        save_embeddings(embeddings, labels, "./embeddings/")
+    
+    4. Load embeddings from disk:
+        loaded_embeddings, loaded_labels = load_embeddings("./embeddings/")
+
+Note:
+    Ensure the DataLoader provided to the extraction functions outputs batches 
+    in the form (data, labels). Make sure to match the output_layer_name in the ONNX extraction 
+    with the appropriate output layer's name from the ONNX model.
+
+Dependencies:
+    - torch
+    - torchvision
+    - onnxruntime
+    - onnx
+"""
+
+import torch
+import torchvision
+import onnxruntime as ort
+from typing import Tuple, Union
+
+def extract_embeddings(model: torch.nn.Module, 
+                       data_loader: torch.utils.data.DataLoader) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    Extract embeddings from the given PyTorch model.
+    """
+    embeddings = []
+    labels = []
+
+    with torch.no_grad():
+        for images, batch_labels in data_loader:
+            outputs = model(images)
```

Comment:
Would it make sense to also specify the output layer / module there? (for example, the module name when you print the model architecture). I could see this being beneficial, but I am also fine with just passing a model that is returning the embeddings directly.